### ISSUE TITLE
Access “Shared with me” folder

### ISSUE THREAD

**Issue by pmjohann** [odeke-em/drive#1076]
Hi,

First thank you for maintaining the project, it’s an awesome tool!
Is there an easy way to be able to pull files from and/or copy them to “My Drive” from the “Shared with me” folder?

Thanks, keep up the good work!

**Comment by otheus** [comment]
I am here to leave the same comment/question. 

Further, while `drive ls -shared` works, `drive ls -id -shared` does not. If I do a copy of a shared resource by ID, it "processes", exits with 0, but does nothing. If I do a copy of the shared resource by name, without the ID, I get "no drive context found; run drive init ..." which doesn't really make sense.

**Comment by odeke-em** [comment]
Hello @pmjohann and @otheus, and welcome to the drive project!

>Is there an easy way to be able to pull files from and/or copy them to “My Drive” from the “Shared with me” folder?

Yes, but first you've manually got to add the shared files to your Drive, otherwise the API can't access them. This is also documented by Google Drive at https://support.google.com/drive/answer/2375057?co=GENIE.Platform%3DDesktop&hl=en
![Screen Shot 2019-12-01 at 12 15 53 PM](https://user-images.githubusercontent.com/4898263/69919672-567b5680-1434-11ea-8e22-8e770f360eda.png)


>Further, while drive ls -shared works, drive ls -id -shared does not.

Doing `ls -id` requires that you pass in the id of the directory you want to list.

>If I do a copy of the shared resource by name, without the ID, I get "no drive context found; run drive init ..." which doesn't really make sense.

Would be useful if you could show me that last step and how to reproduce it.

Google Drive is quite awkward when it comes to shared files and they have be added to your drive for the API to access them.

**Comment by srbrodzik** [comment]
If you add a 'Shared with me' directory to 'My Drive', does it physically move the directory or just create a link?  The reason I ask is that I have a 'Shared with me' directory that is constantly being added to and I want to be able to use 'drive' to access files as they are added.  Will that work?

**Comment by gokulsoumya** [comment]
@srbrodzik Yes it will work. The shared folder in `My Drive` will get updated as the original shared folder changes.
